addons/web/static/src/js/views/calendar/calendar_renderer.js

```diff
@@ -167,6 +168,208 @@ var SidebarFilter = Widget.extend(FieldManagerMixin, {
+            });
+        }
+        /**
+         * Create a month calendar for the date `monthDate` and mount it on container.
+         *
+         * @private
+         * @param {HTMLElement} container
+         * @param {Date} monthDate
+         */
+        _createMonthCalendar(container, monthDate) {
+            return new Calendar(container, Object.assign({}, this.context.options, {
+                defaultDate: monthDate,
+                defaultView: 'dayGridMonth',
+                header: { left: false, center: 'title', right: false },
+                titleFormat: { month: 'short', year: 'numeric' },
+                height: 0,
+                contentHeight: 0,
+                weekNumbers: false,
+                showNonCurrentDates: false,
+                views: {
+                    dayGridMonth: {
+                        columnHeaderText: (date) => moment(date).format("ddd")[0],
+                    },
+                },
+                selectMinDistance: 5, // needed to not trigger select when click
+                dateClick: this._onYearDateClick.bind(this),
+                datesRender: undefined,
+                events: (info, successCB) => {
+                    successCB(this.events);
+                },
+            }));
+        }
+
+        //----------------------------------------------------------------------
+        // Handlers
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         * @param {*} info 
+         */
+        _onYearDateClick(info) {
+            const calendar = this.context.calendar;
+            const events = Object.values(this.events)
+                .filter(event => {
+                    const startUTC = calendar.dateEnv.toDate(event._instance.range.start);
+                    const endUTC = calendar.dateEnv.toDate(event._instance.range.end);
+                    const start = moment(startUTC);
+                    const end = moment(endUTC);
+                    const inclusivity = start.isSame(end, 'day') ? '[]' : '[)';
+                    return moment(info.date).isBetween(start, end, 'day', inclusivity);
+                })
+                .map(event => {
+                    return Object.assign({}, event._def, event._instance.range);
+                });
+            const yearDateInfo = Object.assign({}, info, {
+                view: this,
+                monthView: info.view,
+                events,
+                selectable: this.context.options.selectable,
+            });
+            calendar.publiclyTrigger('yearDateClick', [yearDateInfo]);
+        }
+    }
+    return FullCalendar.createPlugin({
+        views: {
+            dayGridYear: {
+                class: YearView,
+                duration: { years: 1 },
+                defaults: {
+                    fixedWeekCount: true,
+                },
+            },
+        },
+    });
+}
```

Comment:
maybe put this in another file? Like src/js/libs/fullcalendar.js (this is where we put extensions/customizations/overrides of libs)

Reply:
(the file doesn't exist yet, you'd need to create it)